### PR TITLE
Fixed #24497 -- Added Widget.supports_microseconds attribute

### PR DESCRIPTION
**Pull request by wdmgsm**

### CODE REVIEW

File: tests/forms_tests/tests/test_forms.py [hunk 1]
```diff
@@ -1455,15 +1455,37 @@ def delayed_now():
         def delayed_now_time():
             return now.time()
 
+        class HiddenInputWithMicrosec(HiddenInput):
+            supports_microseconds = True
```
Comment (claudep) [inline tests/forms_tests/tests/test_forms.py]: As `supports_microseconds` defaults to True, I don't think we need to create a subclass here, let's just test `HiddenInput`. Same for `TextInput` below.

**Comment by timgraham** [comment]
buildbot, test this please.

**Comment by timgraham** [comment]
Do you think it's safe to change `getattr(self.field.widget, 'supports_microseconds', True)` to `self.field.widget.supports_microseconds` with this commit?
 https://github.com/django/django/blob/e7e39d32fd3aa53cf2861f083c24f9d1b38572d6/django/forms/forms.py#L623
Besides that it looks okay to me.

**Comment by claudep** [comment]
> Do you think it's safe to change getattr(self.field.widget, 'supports_microseconds', True) to self.field.widget.supports_microseconds with this commit?

Yes, I think so.

**Comment by timgraham** [comment]
merged in 4dcc6493418c78db07761180bf6265f5b2bbccbf, thanks!

**Pull request closed without merging**